modules/azure/src/main/java/org/testcontainers/azure/AzuriteContainer.java

```diff
@@ -0,0 +1,175 @@
+ * <p>
+ * Exposed ports:
+ * <ul>
+ *     <li>10000 (blob port)</li>
+ *     <li>10001 (queue port)</li>
+ *     <li>10002 (table port)</li>
+ * </ul>
+ */
+public class AzuriteContainer extends GenericContainer<AzuriteContainer> {
+
+    private static final String DEFAULT_HOST = "127.0.0.1";
+
+    private static final String ALLOW_ALL_CONNECTIONS = "0.0.0.0";
+
+    private static final int DEFAULT_BLOB_PORT = 10000;
+
+    private static final int DEFAULT_QUEUE_PORT = 10001;
+
+    private static final int DEFAULT_TABLE_PORT = 10002;
+
+    private static final String CONNECTION_STRING_FORMAT =
+        "DefaultEndpointsProtocol=%s;AccountName=%s;AccountKey=%s;BlobEndpoint=%s://%s:%d/%s;QueueEndpoint=%s://%s:%d/%s;TableEndpoint=%s://%s:%d/%s;";
+
+    /**
+     * The account name of the default credentials.
+     */
+    private static final String WELL_KNOWN_ACCOUNT_NAME = "devstoreaccount1";
+
+    /**
+     * The account key of the default credentials.
+     */
+    private static final String WELL_KNOWN_ACCOUNT_KEY =
+        "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse(
+        "mcr.microsoft.com/azure-storage/azurite"
+    );
+
+    private String host = DEFAULT_HOST;
+
+    private MountableFile cert = null;
+
+    private String certExtension = null;
+
+    private MountableFile key = null;
+
+    private String pwd = null;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzuriteContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        withExposedPorts(DEFAULT_BLOB_PORT, DEFAULT_QUEUE_PORT, DEFAULT_TABLE_PORT);
+    }
+
+    /**
+     * Configure SSL with a custom certificate and password.
+     *
+     * @param pfxCert  The PFX certificate file
+     * @param password The password securing the certificate
+     * @return this
+     */
+    public AzuriteContainer withSsl(final MountableFile pfxCert, final String password) {
+        cert = pfxCert;
+        pwd = password;
+        certExtension = ".pfx";
+        return this;
+    }
+
+    /**
+     * Configure SSL with a custom certificate and private key.
+     *
+     * @param pemCert The PEM certificate file
+     * @param pemKey  The PEM key file
+     * @return this
+     */
+    public AzuriteContainer withSsl(final MountableFile pemCert, final MountableFile pemKey) {
+        cert = pemCert;
+        key = pemKey;
+        certExtension = ".pem";
+        return this;
+    }
+
+    /**
+     * Sets the hostname we want to use to connect to our emulator. (default: {@link #DEFAULT_HOST})
+     *
+     * @param host The host name
+     * @return this
+     */
+    public AzuriteContainer withHost(final String host) {
+        this.host = host;
+        return this;
+    }
```

Comment:
I can not find any reason why the host should be set. See the other comment I left in `getConnectionString`

Reply:
Please kindly reply to my concerns on that thread to help us resolve this.